### ISSUE TITLE
trainning details

### ISSUE THREAD

**Issue by ooochen-30** [tvaranka/fineface#6]
recently i try to train my own model follow your method, however my model is difficult to converge, and the loss decreases slowly or even does not decrease. can you provide more details about your work, such as the trainning resources，epoch, using time and so on.

**Comment by ooochen-30** [comment]
besides when i read the supplementary material show in the following image, i cannot understand your setup of experiment, at the beginning i think you train two model, one provides text prompt and au as condition, and the other one provides text prompt, image prompt and au as condition. but at here is that mean you train image-adapter and au-adapter separately then use them by add together, or does it have other meanings? i will really appreciated for your answer
![image](https://github.com/user-attachments/assets/5266aeab-54c7-41de-aa9a-d7aad086eee6)

**Comment by tvaranka** [comment]
I will add more details regarding the experimental settings to the supplementary and update it. I'll also update you here on the settings later this week.

My writing is confusing here, sorry about that. What I meant to say is that the adapters are trained separately, but the image adapter is an off-the-shelf adapter and we do not train it. We only train the AU-adapter and integrate it with the already existing IP-adapter with images. During training the AU-adapter, there is no IP-adapter present. This is the same trained AU-adapter that is trained for (text, AU)-to-image purposes.

**Comment by ooochen-30** [comment]
Thank you for your answer and looking forward to your supplementary! 
I am so sorry but still want to ask several questions about distribution smoothing in your paper, where you said that "This approach is particularly relevant in practical scenarios where AUs are annotated by humans and are likely to contain imperfections, particularly since the labels must be provided as integer values. ..." why the labels must be provided as integer values? when you add a small perturbation to the labels, did you perturb all 12 labels of the same sample?

**Comment by tvaranka** [comment]
Not a problem, happy to help. I'll modify that bit, the official FACS is defined by integers and annotaters are trained with only integers. What I meant to say is that the AUs must be provided as integers to the model as there are no other options as of now.

Yes, all of the labels are perturbed. I generated a random of length 12 vector using a sigma that is then added to the sample. This also helps to match the distribution with the automatically annotated part of the dataset.

**Comment by ooochen-30** [comment]
The annotation of LibreFace does not seem to have a intensity of [0,5]. Does this correspond to the statement in the supplementary materials that "samples with a total intensity of less than 0.2 sum of all AUs are filtered"? So, how do you match or unify the automatic annotation results(AffectNet) with the manually annotated intensity(DISFA)

**Comment by tvaranka** [comment]
LibreFace provides the scores in [0, 1], but since it is trained on DISFA, you can simply multiple the values by 5 to get the corresponding range [0, 5]. I will double check to confirm whether this step is performed before or after the filtering. The filtering is performed because LibreFace tends to give small values regardless if an AU is present or not.

**Comment by ooochen-30** [comment]
> LibreFace provides the scores in [0, 1], but since it is trained on DISFA, you can simply multiple the values by 5 to get the corresponding range [0, 5]. I will double check to confirm whether this step is performed before or after the filtering. The filtering is performed because LibreFace tends to give small values regardless if an AU is present or not.

Great! I got it! Thank you for always answering my questions. I think your method is very ingenious, and I hope I can successfully reproduce it.
I also look forward to your trainning details and codes.

**Comment by ooochen-30** [comment]
> > LibreFace provides the scores in [0, 1], but since it is trained on DISFA, you can simply multiple the values by 5 to get the corresponding range [0, 5]. I will double check to confirm whether this step is performed before or after the filtering. The filtering is performed because LibreFace tends to give small values regardless if an AU is present or not.
> 
> Great! I got it! Thank you for always answering my questions. I think your method is very ingenious, and I hope I can successfully reproduce it. I also look forward to your trainning details and codes.

Actually, I tried to imitate your paper to design training code and run it (some details may have been ignored or there were errors), but my model's loss decreased very slowly or even stagnated. I felt confused and repeatedly read Fineface, but most of the time I didn't know where to modify...

**Comment by tvaranka** [comment]
For training I used two MI250Xs, the training took rougly 7 hours, 20k steps with batch size of 12. The loss stays very stagnant as we want to avoid overfitting. The filtering of 0.2 is done prior to multiplying by 5.

I am getting closer to finishing the IP-adapter & AU pipeline. I also have the data part finished soon. After that I will work on getting the training code ready. I appreciate your patience. 

In the meantime if you have any more questions happy to help @ooochen-30

**Comment by ooochen-30** [comment]
hello,i back again. Recently i train the vanilla model(no distribution smooth,no special designed encoder and other details i may ignore), and the loss is stagnant too. Suddenly, I want to know what criteria you use to determine the end of a training session or why you choose 20k steps base on this situation. By the way, if I am not mistaken, it seems that in the provided demo, you did not apply distribution smoothing and quantify with a 20% probability during inference, as stated in the supplementary materials of the paper.

**Comment by tvaranka** [comment]
I chose 20k steps based on qualitative results since the loss was relatively stagnant. I would generate samples every 1k steps and observed that around 20k steps was sufficiently good. You may find better results at different steps, this was a somewhat arbitrary decision.

Let me clarify the last sentence from the first paragraph in the supplementary. The 20% probability of quantization is done during training. This is because typically during inference you will want to use integer values (for simplicity), but because of the distribution smoothing none of the values are integers during training. This would then create a discrepancy between the training and inference. Therefore for 20% of the time the labels are quantized during training. This does not require any further actions during inference.

**Comment by ooochen-30** [comment]
Okay, I understand what you mean. It's a very natural and reasonable design. :+1:

**Comment by tvaranka** [comment]
Added training script
